src/main/java/net/mcreator/ui/component/JStringListField.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component;
+
+import net.mcreator.ui.component.util.ComponentUtils;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.dialogs.ListEditorDialog;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.init.UIRES;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+
+import javax.annotation.Nullable;
+import javax.swing.*;
+import java.awt.*;
+import java.awt.datatransfer.StringSelection;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * This class represents a component that stores list of strings and allows to edit them using {@link ListEditorDialog}.
+ */
+public class JStringListField extends JPanel {
+
+	private final List<String> textList = new ArrayList<>();
+	private int index = -1;
+	private boolean joinEntries = false, uniqueEntries = false;
+
+	private final JLabel label = new JLabel() {
+		@Override public String getToolTipText() {
+			return joinEntries || index < 0 ? null : L10N.t("components.string_list.item", index + 1);
+		}
+	};
+
+	private final TechnicalButton edit = new TechnicalButton(UIRES.get("16px.edit.gif")) {
+		@Override public String getName() {
+			return "TechnicalButton";

```

Comment:
Why does this button need to return TechnicalButton and override the parent method? Seems like something is not ok if this is needed and needs improvement, as this is really strange looking to me right now

Reply:
✔️ Fixed